pkg/util/maven/maven_proxies.go

```diff
@@ -0,0 +1,89 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package maven
+
+import (
+	"net/url"
+	"os"
+	"strings"
+)
+
+var ProxyFromEnvironment = proxyFromEnvironment{}
+
+type proxyFromEnvironment struct{}
+
+func (proxyFromEnvironment) apply(settings *Settings) error {
+	if httpProxy := os.Getenv("HTTP_PROXY"); httpProxy != "" {
+		proxy, err := parseProxyFromEnvVar(httpProxy)
+		if err != nil {
+			return err
+		}
+		proxy.ID = "http-proxy"
+		settings.Proxies = append(settings.Proxies, proxy)
+	}
+
+	if httpsProxy := os.Getenv("HTTPS_PROXY"); httpsProxy != "" {
```

Comment:
Nit: duplicated code could benefit a refactoring into a function

Reply:
Sure, let's do this in a subsequent PR so as to have this one in the upcoming release.